TensorFlowGraphOperation.swift

```diff
@@ -0,0 +1,168 @@
+
+  func execute<T0 : TensorArrayProtocol, T1 : TensorArrayProtocol>(
+    _ count0: Int,
+    _ count1: Int
+  ) -> (T0, T1)
+
+  func execute<T0 : TensorArrayProtocol, T1 : TensorArrayProtocol, T2 : TensorArrayProtocol>(
+    _ count0: Int,
+    _ count1: Int,
+    _ count2: Int
+  ) -> (T0, T1, T2)
+
+  func execute<T0 : TensorArrayProtocol, T1 : TensorArrayProtocol, T2 : TensorArrayProtocol, T3 : TensorArrayProtocol>(
+    _ count0: Int,
+    _ count1: Int,
+    _ count2: Int,
+    _ count3: Int
+  ) -> (T0, T1, T2, T3)
+
+  func execute<T0 : TensorArrayProtocol, T1 : TensorArrayProtocol, T2 : TensorArrayProtocol, T3 : TensorArrayProtocol, T4 : TensorArrayProtocol>(
+    _ count0: Int,
+    _ count1: Int,
+    _ count2: Int,
+    _ count3: Int,
+    _ count4: Int
+  ) -> (T0, T1, T2, T3, T4)
+
+  func execute<T0 : TensorArrayProtocol, T1 : TensorArrayProtocol, T2 : TensorArrayProtocol, T3 : TensorArrayProtocol, T4 : TensorArrayProtocol, T5 : TensorArrayProtocol>(
+    _ count0: Int,
+    _ count1: Int,
+    _ count2: Int,
+    _ count3: Int,
+    _ count4: Int,
+    _ count5: Int
+  ) -> (T0, T1, T2, T3, T4, T5)
+
+  func execute<T0 : TensorArrayProtocol, T1 : TensorArrayProtocol, T2 : TensorArrayProtocol, T3 : TensorArrayProtocol, T4 : TensorArrayProtocol, T5 : TensorArrayProtocol, T6 : TensorArrayProtocol>(
+    _ count0: Int,
+    _ count1: Int,
+    _ count2: Int,
+    _ count3: Int,
+    _ count4: Int,
+    _ count5: Int,
+    _ count6: Int
+  ) -> (T0, T1, T2, T3, T4, T5, T6)
+    
+  func execute<T0 : TensorArrayProtocol, T1 : TensorArrayProtocol, T2 : TensorArrayProtocol, T3 : TensorArrayProtocol, T4 : TensorArrayProtocol, T5 : TensorArrayProtocol, T6 : TensorArrayProtocol, T7 : TensorArrayProtocol>(
+    _ count0: Int,
+    _ count1: Int,
+    _ count2: Int,
+    _ count3: Int,
+    _ count4: Int,
+    _ count5: Int,
+    _ count6: Int,
+    _ count7: Int
+  ) -> (T0, T1, T2, T3, T4, T5, T6, T7)
+
+  func execute<T0 : TensorArrayProtocol, T1 : TensorArrayProtocol, T2 : TensorArrayProtocol, T3 : TensorArrayProtocol, T4 : TensorArrayProtocol, T5 : TensorArrayProtocol, T6 : TensorArrayProtocol, T7 : TensorArrayProtocol, T8 : TensorArrayProtocol>(
+    _ count0: Int,
+    _ count1: Int,
+    _ count2: Int,
+    _ count3: Int,
+    _ count4: Int,
+    _ count5: Int,
+    _ count6: Int,
+    _ count7: Int,
+    _ count8: Int
+  ) -> (T0, T1, T2, T3, T4, T5, T6, T7, T8)
+  
+  func execute<T0 : TensorArrayProtocol, T1 : TensorArrayProtocol, T2 : TensorArrayProtocol, T3 : TensorArrayProtocol, T4 : TensorArrayProtocol, T5 : TensorArrayProtocol, T6 : TensorArrayProtocol, T7 : TensorArrayProtocol, T8 : TensorArrayProtocol, T9 : TensorArrayProtocol>(
+    _ count0: Int,
+    _ count1: Int,
+    _ count2: Int,
+    _ count3: Int,
+    _ count4: Int,
+    _ count5: Int,
+    _ count6: Int,
+    _ count7: Int,
+    _ count8: Int,
+    _ count9: Int
+  ) -> (T0, T1, T2, T3, T4, T5, T6, T7, T8, T9)
+
+  // The following should work as well.
```

Comment:
How would these functions decide which instances of `TensorArrayProtocol` to create as their return values?

Reply:
I assume this is using return-type type inference in the generated ops case?

Reply:
As @saeta says.